### ISSUE TITLE
Cache metadata.db for search/list --available

### ISSUE THREAD

**Issue by optiz0r** [Sabayon/enman#2]
This operation is slow (takes several seconds) and is carried out every time the tool runs. This should be cached locally with either a command to flush the cache, or an explicit update command to download a new copy of the metadata (like equo update downloads new copies of the repo databases only on request).

**Comment by mudler** [comment]
Yes, it is a good idea. let's use this issue to track the progress and discuss the enhancement.

Technical thoughts:

The cache then should be stored in a format that is readable by the client as fast as possible. The most simplier and portable thing that comes now to my mind is [DB_FILE](http://docstore.mik.ua/orelly/perl/cookbook/ch14_02.htm) / [tie() ](http://perldoc.perl.org/functions/tie.html) which is available in the Perl core and quite fast.
